dspace-api/src/main/java/org/dspace/content/dao/impl/CommunityDAOImpl.java

```diff
@@ -115,42 +126,29 @@ public Community findByAdminGroup(Context context, Group group) throws SQLExcept
                         "  resourcepolicy.resource_type_id = 4 AND eperson.eperson_id = ?", context.getCurrentUser()
                         .getID());
         */
-        Criteria criteria = createCriteria(context, Community.class);
-        criteria.createAlias("resourcePolicies", "resourcePolicy");
 
-        Disjunction actionQuery = Restrictions.or();
+        CriteriaBuilder criteriaBuilder = getCriteriaBuilder(context);
+        CriteriaQuery criteriaQuery = getCriteriaQuery(criteriaBuilder, Community.class);
+        Root<Community> communityRoot = criteriaQuery.from(Community.class);
+        Join<Community, ResourcePolicy> join = communityRoot.join("resourcePolicies");
+        List<Predicate> orPredicates = new LinkedList<Predicate>();
```

Comment:
Couldn't this just be 'Predicate[] orPredicate = new Predicate[actions.length]'?  Then we don't have to create more work and more garbage by calculating toArray() later.  IIRC there's similar code in CollectionDAOImpl.

Reply:
That's a good point. That should be possible at first sight and there are indeed multiple cases where this could be implemented like you proposed.